src/config/buffer-config.c

```diff
@@ -36,78 +37,145 @@ int Read_ClientBuffer(XML_NODE node, __attribute__((unused)) void *d1, void *d2)
             if (logr->buflength <= 0 || logr->buflength > 100000) {
-                merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                if (output == NULL){
+                    merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                } else {
+                    snprintf(message, OS_FLSIZE,
+                        "Invalid value for element '%s': %s.",
+                        node[i]->element, node[i]->content);
+                    wm_strcat(output, message, '\n');
+                }
                 return (OS_INVALID);
             }
 
         } else if (strcmp(node[i]->element, xml_events_per_second) == 0) {
             if (!OS_StrIsNum(node[i]->content)) {
-                merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                if (output == NULL){
+                    merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                } else {
+                    snprintf(message, OS_FLSIZE,
+                        "Invalid value for element '%s': %s.",
+                        node[i]->element, node[i]->content);
+                    wm_strcat(output, message, '\n');
+                }
                 return (OS_INVALID);
             }
             logr->events_persec = atoi(node[i]->content);
             if (logr->events_persec <= 0 || logr->events_persec > 1000) {
-                merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                if (output == NULL){
+                    merror(XML_VALUEERR, node[i]->element, node[i]->content);
+                } else {
+                    snprintf(message, OS_FLSIZE,
+                        "Invalid value for element '%s': %s.",
+                        node[i]->element, node[i]->content);
+                    wm_strcat(output, message, '\n');
+                }
                 return (OS_INVALID);
             }
 
-        } else {
+        } else if (output == NULL){
             merror(XML_INVELEM, node[i]->element);
             return (OS_INVALID);
+        } else {
+            snprintf(message, OS_FLSIZE,
+                "Invalid element in the configuration: '%s'.",
+                node[i]->element);
+            wm_strcat(output, message, '\n');
+            return (OS_INVALID);
         }
         i++;
     }
 
     return (0);
 }
 
-int Test_ClientBuffer(const char * path){
-    int fail = 0;
+int Test_ClientBuffer(const char *path, int type, char **output){
     agent test_clientBuffer = { .server = 0 };
 
-    if (ReadConfig(CAGENT_CONFIG | CBUFFER, path, NULL, &test_clientBuffer) < 0) {
-		merror(RCONFIG_ERROR,"ClientBuffer", path);
-		fail = 1;
+    if (ReadConfig(CBUFFER | type, path, NULL, &test_clientBuffer, output) < 0) {
+        if (output == NULL){
+            merror(CONF_READ_ERROR, "ClientBuffer");
+        } else{
+            wm_strcat(output, "ERROR: Invalid configuration in ClientBuffer", '\n');
+        }
+        Free_Client(&test_clientBuffer);
```

Comment:
This sentence is the same as line 179.  Please, modify it so that it doesn't repeat itself.